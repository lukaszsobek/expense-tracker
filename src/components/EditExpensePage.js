import React from "react"

const EditExpensePage = props => {
    console.log(props)
    return <div>Edit ing id {props.match.params.id}</div>
}

export default EditExpensePage