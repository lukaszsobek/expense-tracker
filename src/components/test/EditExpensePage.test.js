import React from "react"
import { shallow } from "enzyme"

import { EditExpensePage } from "../EditExpensePage"

const testExpense = {
    description: "testDesc",
    note: "testNote",
    amount: 200,
    createdAt: 0,
}

describe("EditExpensePage", () => {

    let editExpenseSpy, removeExpenseSpy, historySpy, wrapper

    beforeEach(() => {
        editExpenseSpy = jest.fn()
        removeExpenseSpy = jest.fn()
        historySpy = {
            push: jest.fn()
        }
        wrapper = shallow(
            <EditExpensePage
                editedExpense={testExpense}
                editExpense={editExpenseSpy}
                removeExpense={removeExpenseSpy}
                history={historySpy}
            />
        )
    })

    it("renders Component with data",() => {
        expect(wrapper).toMatchSnapshot()
    })

    it("removes item on click",() => {
        wrapper.find("button").simulate("click")
        expect(removeExpenseSpy).toBeCalledWith(testExpense)
        expect(historySpy.push).toBeCalledWith("/")
    })

    it("submits item",() => {
        wrapper.find("ExpenseForm").prop("onSubmit")(testExpense)
        expect(editExpenseSpy).toBeCalledWith(testExpense.id,testExpense)
        expect(historySpy.push).toBeCalledWith("/")
    })


})