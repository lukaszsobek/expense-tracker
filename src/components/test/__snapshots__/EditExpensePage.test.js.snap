// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EditExpensePage renders Component with data 1`] = `
<div>
  <div
    className="page-header"
  >
    <div
      className="app-container"
    >
      Edit Expense
    </div>
  </div>
  <ExpenseForm
    editedExpense={
      Object {
        "amount": 200,
        "createdAt": 0,
        "description": "testDesc",
        "note": "testNote",
      }
    }
    onSubmit={[Function]}
  />
  <button
    className="expense-form__remove-button"
    onClick={[Function]}
  >
    Remove
  </button>
</div>
`;
